Software/src/LedDeviceAdalight.cpp

```diff
@@ -246,13 +248,30 @@ void LedDeviceAdalight::open()
 	emit openDeviceSuccess(ok);
 }
 
+void LedDeviceAdalight::writeLastWill()
+{
+	if (m_AdalightDevice->bytesToWrite() == 0) {
+		DEBUG_MID_LEVEL << Q_FUNC_INFO << "Writing last will frame";
+		setColors(m_colorsSaved);
+	}
+}
+
 bool LedDeviceAdalight::writeBuffer(const QByteArray & buff)
 {
 	DEBUG_MID_LEVEL << Q_FUNC_INFO << "Hex:" << buff.toHex();
 
 	if (m_AdalightDevice == NULL || m_AdalightDevice->isOpen() == false)
 		return false;
 
+	if (m_AdalightDevice->bytesToWrite() > 0) {
+		DEBUG_MID_LEVEL << Q_FUNC_INFO << "Serial bytesToWrite:" << m_AdalightDevice->bytesToWrite() << ", skipping current frame";
+		// If no more writes will be done ("Send data only of colors changed")
+		// re-schedule last skipped frame in case it's important (for ex a black frame to turn off)
+		m_lastWillTimer->start(100);
+		return true;
+	}
+	m_lastWillTimer->stop();
```

Comment:
close() should probably check for the last will timer / write the last will in any case

Reply:
Good idea, added. I'm just not 100% sure what is exactly happening to the internal buffer when `close()` is called: is it discarded or does `close()` wait til it's empty...
[This](https://doc.qt.io/qt-5/qiodevice.html#aboutToClose) suggests the latter, but as I saw while messing with this you can write quite a bit of data which can take few seconds to write out